src/packet/packet_structure.rs

```diff
@@ -176,6 +176,61 @@ impl Packet {
     }
 }
 
+/// This packet type have similar properties to `Packet` except that it doesn't own anything, and additionally has `PacketType`.
+#[derive(Debug)]
+pub struct GenericPacket<'a> {
+    /// defines packet type, currently `Fragment` type is not supported
+    pub(crate) packet_type: PacketType,
+    /// the raw payload of the packet
+    pub(crate) payload: &'a [u8],
+    /// defines on how the packet will be delivered.
+    pub(crate) delivery: DeliveryGuarantee,
+    /// defines on how the packet will be ordered.
+    pub(crate) ordering: OrderingGuarantee,
+}
+
+impl<'a> GenericPacket<'a> {
+    /// Creates a connection manager specific packet
+    pub fn manager_packet(
```

Comment:
manager packet sounds weird, maybe we should change that name to connection packet